src/dictmap.rs

```diff
@@ -0,0 +1,31 @@
+// Convenient alias to build IndexMap using a custom hasher.
+// For the moment, we use ahash which is the default hasher
+// for hashbrown::HashMap, another hashmap we use
+
+pub type DictMap<K, V> = indexmap::IndexMap<K, V, ahash::RandomState>;
+
+#[macro_export]
+macro_rules! _dictmap_new {
```

Comment:
A macro works fine here. In case we want to keep the common syntax `DictMap::new()` we could define a custom trait like 
```rust
pub trait Init {
    fn new() -> Self
    where
        Self: Sized;

    fn with_capacity(n: usize) -> Self
    where
        Self: Sized;
}

impl<K, V> Init for DictMap<K, V> {
    fn new() -> Self {
        indexmap::IndexMap::with_capacity_and_hasher(
            0,
            ahash::RandomState::default(),
        )
    }

    fn with_capacity(n: usize) -> Self {
        indexmap::IndexMap::with_capacity_and_hasher(
            n,
            ahash::RandomState::default(),
        )
    }
}
```

Reply:
I like this idea and I implemented it. One remarkable thing to say is that the trait must be in scope, otherwise we get:
```
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use crate::dictmap::Init;`
```
To overcome that, I replaced the imports with `use crate::dictmap::*` to guarantee the trait would be in scope